pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -414,6 +416,41 @@ private void closeLeaderElectionService() throws Exception {
         }
     }
 
+    private boolean isManagedLedgerNotFoundException(Throwable e) {
+        Throwable realCause = e.getCause();
+        return realCause instanceof ManagedLedgerException.MetadataNotFoundException
+                || realCause instanceof MetadataStoreException.NotFoundException;
+    }
+
+    private void deleteHeartbeatResource() {
+        if (this.brokerService != null) {
+            LOG.info("forcefully delete heartbeat topic when close broker");
+
+            String heartbeatTopicNameV1 = getHeartbeatTopicName(getAdvertisedAddress(), getConfiguration(), false);
+            String heartbeatTopicNameV2 = getHeartbeatTopicName(getAdvertisedAddress(), getConfiguration(), true);
```

Comment:
I don't think that `getAdvertisedAddress()` should be passed here. In BrokersBase.internalRunHealthCheck, it's using `getBrokerId()` instead. Please switch to use that here too.

Reply:
done.